src/libstd/process.rs

```diff
@@ -0,0 +1,837 @@
+    ///
+    /// Note that environment variable names are case-insensitive (but case-preserving) on Windows,
+    /// and case-sensitive on all other platforms.
+    pub fn env<S: ?Sized, T: ?Sized>(&mut self, key: &S, val: &T) -> &mut Command where
+        S: AsOsStr, T: AsOsStr
+    {
+        self.inner.env(key.as_os_str(), val.as_os_str());
+        self
+    }
+
+    /// Removes an environment variable mapping.
+    pub fn env_remove<S: ?Sized + AsOsStr>(&mut self, key: &S) -> &mut Command {
+        self.inner.env_remove(key.as_os_str());
+        self
+    }
+
+    /// Clears the entire environment map for the child process.
+    pub fn env_clear(&mut self) -> &mut Command {
+        self.inner.env_clear();
+        self
+    }
+
+    /// Set the working directory for the child process.
+    pub fn current_dir<P: AsPath + ?Sized>(&mut self, dir: &P) -> &mut Command {
+        self.inner.cwd(dir.as_path().as_os_str());
+        self
+    }
+
+    /// Configuration for the child process's stdin handle (file descriptor 0).
+    /// Defaults to `CreatePipe(true, false)` so the input can be written to.
+    pub fn stdin(&mut self, cfg: Stdio) -> &mut Command {
+        self.stdin = Some(cfg.0);
+        self
+    }
+
+    /// Configuration for the child process's stdout handle (file descriptor 1).
+    /// Defaults to `CreatePipe(false, true)` so the output can be collected.
+    pub fn stdout(&mut self, cfg: Stdio) -> &mut Command {
+        self.stdout = Some(cfg.0);
+        self
+    }
+
+    /// Configuration for the child process's stderr handle (file descriptor 2).
+    /// Defaults to `CreatePipe(false, true)` so the output can be collected.
+    pub fn stderr(&mut self, cfg: Stdio) -> &mut Command {
+        self.stderr = Some(cfg.0);
+        self
+    }
+
+    fn spawn_inner(&self, default_io: StdioImp) -> io::Result<Child> {
+        let (their_stdin, our_stdin) = try!(
+            setup_io(self.stdin.as_ref().unwrap_or(&default_io), 0, true)
+        );
+        let (their_stdout, our_stdout) = try!(
+            setup_io(self.stdout.as_ref().unwrap_or(&default_io), 1, false)
+        );
+        let (their_stderr, our_stderr) = try!(
+            setup_io(self.stderr.as_ref().unwrap_or(&default_io), 2, false)
+        );
+
+        match ProcessImp::spawn(&self.inner, their_stdin, their_stdout, their_stderr) {
+            Err(e) => Err(e),
+            Ok(handle) => Ok(Child {
+                handle: handle,
+                status: None,
+                stdin: our_stdin.map(|fd| ChildStdin { inner: fd }),
+                stdout: our_stdout.map(|fd| ChildStdout { inner: fd }),
+                stderr: our_stderr.map(|fd| ChildStderr { inner: fd }),
+            })
+        }
+    }
+
+    /// Executes the command as a child process, returning a handle to it.
+    ///
+    /// By default, stdin, stdout and stderr are inherited by the parent.
+    pub fn spawn(&self) -> io::Result<Child> {
```

Comment:
Hm, I forgot about this earlier, but if `Stdio` holds an I/O object and needs to transfer ownership that in theory could affect the `&self` here. We may want to consider taking `&mut self`, but on the other hand the worst case scenario would be adding a `RefCell` which wouldn't necessarily be the end of the world.